### ISSUE TITLE
[FEATURE REQUEST] SELinux support for file.directory and file.recurse

### ISSUE THREAD

**Issue by p3lim** [saltstack/salt#60445]
**Is your feature request related to a problem? Please describe.**

#49860 added support for managing SELinux context of files with `file.managed`, I'd like to see this for `file.directory` and `file.recurse` as well.

**Describe the solution you'd like**

Extending #49860 for `file.directory` and `file.recurse`.  
The groundwork has already been done, and extending this to the other states should be trivial.

**Describe alternatives you've considered**

Managing file contexts manually with `cmd.run`, which is far from ideal.

**Comment by sagetherage** [comment]
@p3lim you're correct and giving this a label of `enhancement` only to help delineate visually the ground work is done. We will accept a PR and review for this enhancement from the community and I will attempt to get planning in place around enhancements and how they may be handled differently. Currently, they are like features.
